backend/api/scraper_handler.go

```diff
@@ -0,0 +1,280 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/AustrianDataLAB/GeWoScout/backend/cosmos"
+	"github.com/AustrianDataLAB/GeWoScout/backend/models"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos"
+	"github.com/go-chi/render"
+	"github.com/xeipuuv/gojsonschema"
+	"golang.org/x/text/runes"
+	"golang.org/x/text/transform"
+	"golang.org/x/text/unicode/norm"
+	"log"
+	"net/http"
+	"slices"
+	"strconv"
+	"strings"
+	"time"
+	"unicode"
+)
+
+type bindingInput struct {
+	Data     bindingData     `json:"Data"`
+	Metadata bindingMetadata `json:"Metadata"`
+}
+
+type bindingData struct {
+	Msg string `json:"msg"`
+}
+
+type bindingMetadata struct {
+	Id           string `json:"Id"`
+	DequeueCount string `json:"DequeueCount"`
+}
+
+func (h *Handler) CreateScraperResultHandler() http.HandlerFunc {
+
+	schemaLoader := gojsonschema.NewStringLoader(models.ScraperResultListingSchema)
+	schema, err := gojsonschema.NewSchema(schemaLoader)
+	if err != nil {
+		log.Fatalf("Failed to create schema: %s", err.Error())
+	}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		injectedData := bindingInput{}
+		dec := json.NewDecoder(r.Body)
+		if err := dec.Decode(&injectedData); err != nil {
+			render.Status(r, http.StatusInternalServerError)
+			render.JSON(w, r, models.InvokeResponse{
+				Logs: []string{fmt.Sprintf("ScraperResultHandler | Failed to read invoke request body: %s", err.Error())},
+			})
+			return
+		}
+
+		msgId, err := strconv.Unquote(injectedData.Metadata.Id)
```

Comment:
**suggestion:** This is something we could maybe move to a middleware and keep this id in context. I would handle this in a separate issue, though. What do you think?